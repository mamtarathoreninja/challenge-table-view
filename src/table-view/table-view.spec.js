import React from 'react'
import TableView from './table-view'
import { Select, Table } from '../index'

const minProps = {
  columns: [
    {
      label: 'Name',
      dataKey: 'name'
    },
    {
      label: 'Age',
      dataKey: 'age'
    }
  ],
  rows: [
    { name: 'Mike', age: 25, id: 1245 },
    { name: 'Jennifer', age: 30, id: 46874 }
  ],
  filter: [
    {label: 'Name', dataKey: 'name' }
    ]
}

test('renders a TableView', () => {
  const wrapper = shallow(<TableView {...minProps} />)
  expect(wrapper).toMatchSnapshot()
})

test('should have 1 Select', () => {
  const wrapper = mount(<TableView {...minProps} />)
  expect(wrapper.find(Select)).toHaveLength(1)
})

test('should have 1 table', () => {
  const wrapper = mount(<TableView {...minProps} />)
  expect(wrapper.find(Table)).toHaveLength(1)
})
