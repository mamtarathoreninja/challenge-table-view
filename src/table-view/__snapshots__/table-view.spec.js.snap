// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a TableView 1`] = `
<TableView
  classes={
    Object {
      "flexDiv": "TableView-flexDiv-2",
      "root": "TableView-root-1",
      "table": "TableView-table-3",
    }
  }
  columns={
    Array [
      Object {
        "dataKey": "name",
        "label": "Name",
      },
      Object {
        "dataKey": "age",
        "label": "Age",
      },
    ]
  }
  filter={
    Array [
      Object {
        "dataKey": "name",
        "label": "Name",
      },
    ]
  }
  rows={
    Array [
      Object {
        "age": 25,
        "id": 1245,
        "name": "Mike",
      },
      Object {
        "age": 30,
        "id": 46874,
        "name": "Jennifer",
      },
    ]
  }
/>
`;
